test/uk/gov/hmrc/helptosave/repo/UserCapStoreSpec.scala

```diff
@@ -50,23 +38,19 @@ class UserCapStoreSpec extends TestSupport with MongoTestSupport[UserCap, MongoU
     "getting the user-cap" should {
 
       "return the existing record successfully" in {
-
-        mockGet()(toFuture(Some(record)))
-        Await.result(mongoStore.get(), 5.seconds) shouldBe Some(record)
+        withMongo { reactiveMongoComponent ⇒
+          val store = newUserCapMongoStore(reactiveMongoComponent)
+          Await.result(store.doUpdate(record), 5.seconds)
+          Await.result(store.get(), 5.seconds) shouldBe Some(record)
+        }
       }
 
       "returns None if no record exists" in {
-        mockGet()(toFuture(None))
-        Await.result(mongoStore.get(), 5.seconds) shouldBe None
-      }
-    }
-
-    "updating the user-cap" should {
-
-      "return the updated record successfully" in {
-
-        mockUpdate(record)(Right(Some(record)))
-        Await.result(mongoStore.upsert(record), 5.seconds) shouldBe Some(record)
+        withMongo { reactiveMongoComponent ⇒
+          val store = newUserCapMongoStore(reactiveMongoComponent)
+          store.removeAll()
+          Await.result(store.get(), 5.seconds) shouldBe None
```

Comment:
check if you can define a method in TestSupport something like 
``` def await[T](f: Future[T]): T = {
        Await.result(f, 5.seconds)
      } `` so that you can use it every where

Reply:
I tried but the tests resulted in futures not complete